### PR TITLE
Update Autorest to 3.0.0-beta.20211112.1

### PR DESCRIPTION
**Pull request by AlexanderSher**
Two main changes:
- Subclient support
- Rename Get to Get<resource_name> automatically

### CODE REVIEW

File: sdk/template-LLC/Azure.Template.LLC/api/Azure.Template.LLC.netstandard2.0.cs [hunk 1]
```diff
@@ -9,10 +9,10 @@ public partial class TemplateServiceClient
...
         public virtual Azure.Response Delete(string resourceId, Azure.RequestContext context = null) { throw null; }
         public virtual System.Threading.Tasks.Task<Azure.Response> DeleteAsync(string resourceId, Azure.RequestContext context = null) { throw null; }
-        public virtual Azure.Response Get(string resourceId, Azure.RequestContext context = null) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.Response> GetAsync(string resourceId, Azure.RequestContext context = null) { throw null; }
         public virtual Azure.Pageable<System.BinaryData> GetResources(Azure.RequestContext context = null) { throw null; }
         public virtual Azure.AsyncPageable<System.BinaryData> GetResourcesAsync(Azure.RequestContext context = null) { throw null; }
+        public virtual Azure.Response GetTemplateService(string resourceId, Azure.RequestContext context = null) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response> GetTemplateServiceAsync(string resourceId, Azure.RequestContext context = null) { throw null; }
```
Comment (fengzhou-msft) [inline sdk/template-LLC/Azure.Template.LLC/api/Azure.Template.LLC.netstandard2.0.cs:15]: We may want to use directive to rename the get method to `GetResource` based on the list method name `GetResources` ([swagger](https://github.com/Azure/azure-sdk-for-net/blob/3d0a29db2a3e50b302c5aa8d1ab3ed2081afe366/sdk/template-LLC/Azure.Template.LLC/src/swagger/swagger.json)). The original operation Id is `Get`, without an operation group, the generator uses the client name `TemplateService`, but it's not the right guess in this RP.

**Review by christothes — APPROVED** [review]
confidentialledger looks ok to me.

File: sdk/template-LLC/Azure.Template.LLC/api/Azure.Template.LLC.netstandard2.0.cs (hunk 1 above)
Comment (AlexanderSher) [reply sdk/template-LLC/Azure.Template.LLC/api/Azure.Template.LLC.netstandard2.0.cs:15]: Do you want to do it in this PR or we can do it in a separate PR?

Comment (fengzhou-msft) [reply sdk/template-LLC/Azure.Template.LLC/api/Azure.Template.LLC.netstandard2.0.cs:15]: I will update it in another PR.

**Pull request merged**